src/Packetery/Module/Options/FeatureFlagManager.php

```diff
@@ -113,52 +135,41 @@ private function fetchFlags(): array {
 	 * @throws Exception From DateTimeImmutable.
 	 */
 	private function getFlags(): array {
-		static $flags;
-
-		if ( ! isset( $flags ) ) {
-			$flags = get_option( self::FLAGS_OPTION_ID );
+		if ( ! isset( self::$flags ) || null === self::$flags ) {
+			self::$flags = $this->wpAdapter->getOption( self::FLAGS_OPTION_ID );
 		}
 
-		if ( true === get_option( self::DISABLED_DUE_ERRORS_OPTION_ID ) ) {
-			return $flags ? $flags : [];
+		if ( true === $this->wpAdapter->getOption( self::DISABLED_DUE_ERRORS_OPTION_ID ) ) {
+			return self::$flags ? self::$flags : [];
 		}
 
 		$hasApiKey = ( null !== $this->optionsProvider->get_api_key() );
-		if ( false === $flags ) {
+		if ( false === self::$flags ) {
 			if ( ! $hasApiKey ) {
-				$flags = [];
+				self::$flags = [];
 
-				return $flags;
+				return self::$flags;
 			}
 
-			$flags = $this->fetchFlags();
+			self::$flags = $this->fetchFlags();
 
-			return $flags;
+			return self::$flags;
 		}
 
-		if ( $hasApiKey && isset( $flags[ self::FLAG_LAST_DOWNLOAD ] ) ) {
+		if ( $hasApiKey && isset( self::$flags[ self::FLAG_LAST_DOWNLOAD ] ) ) {
```

Comment:
create a private method `isLastDownloadValid()`.

Maybe change the class const VALID_HOURS to `class const VALID_SECOND = 4 * HOUR_IN_SECONDS`.

Then you can use the if statement like this:
```
if ($now-> getTimestamp() > ($lastUpdate-> getTimestamp() + self::VALID_SECOND)
```
or 
```
$validUntil = $lastUpdate->modify('+ self::VALID_SECONDS); 
if ($validUntil < $now) {

```